solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -391,10 +390,6 @@ public String getDataDir() {
     return dataDir;
   }
 
-  public String getUlogDir() {
-    return ulogDir;
-  }
-
```

Comment:
My best understanding is that this was a kind of memoization, to prevent repeated computation of the same string? But it's misleading to have it in here at all, since the only context in which it was used did not actually need the _default_ value (this is available from UpdateLog instance), but rather was interested in consulting the explicit value from CoreDescriptor (often `null`), to properly determine ulog definition hierarchy:
1. from `core.properties`
2. from `updateLog` "dir" config
3. default (`[instanceDir]/data/tlog/`)

... so incorporating the third of the above hierarchy into the `SolrCore.getUlogDir()` abstraction made it useless in the only contexts in which it was actually used.